### ISSUE TITLE
Replace Legacy `Dec` objects

### ISSUE THREAD

**Issue by vitsalis** [babylonchain/babylon#410]
Seems that all `Dec` objects have become legacy. We should investigate what's the proposed replacement and work with it.

**Comment by RafilxTenfen** [comment]
The `LegacyDec` should be replaced by `Int` - [cosmossdk.io/math.Int](https://pkg.go.dev/cosmossdk.io/math#Int)
But, the `LegacyDec` is still being used in a lot of cosmos-sdk core modules like 
- [x/gov](https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/keeper/tally.go#L239)
- [x/mint](https://github.com/cosmos/cosmos-sdk/blob/main/x/mint/types/minter.go#L46)
- [x/staking](https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/types/validator.go#L334)

So, it is going to take some time for them to upgrade. The places where the decimals of dec were being used for defining percentages of slashing could be replaced by an `math.Int` with 6 precision decimals 

- `100%    = 1_000000`
-  `50%     = 0_500000`
-  `10%     = 0_100000`
-  `1%      = 0_010000`
-  `0.1%    = 0_001000`
-  `0.01%   = 0_000100`
-  `0.001%  = 0_000010`
-  `0.0001% = 0_000001`

following this pattern, the least percentage that can be defined would be `0.0001` if more precision is needed it could increase the precision points defined for each case

**Comment by SebastianElvis** [comment]
Yep I still see `LegacyDec` in many places at Cosmos SDK. The main use cases of `LegacyDec` in Babylon are

- calculating the fractions of rewards for BTC timestamping/staking stakeholders in the [incentive](https://github.com/babylonchain/babylon/blob/dev/x/incentive) module
- defining the slashing rate in [BTC staking module](https://github.com/babylonchain/babylon/blob/105216ae6413bf295c85d8bad9b35a2bc7cc1e39/x/btcstaking/types/params.go#L62-L63)

This migration will indeed be a big one. My vague idea is to replace `LegacyNewDec` with `NewIntWithDecimal` for generating decimal values, and migrate all function invocations, etc.. 

This task is at low priority given that Cosmos SDK itself hasn't finished the migration yet, but please feel free to take it
